# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/devil/hotfix-nvtt-2.1.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Fix the build with nvidia-texture-tools 2.1: nvcore headers are no longer
installed and OutputHandler has a new pure virtual endImage().

--- DevIL/src-IL/src/il_nvidia.cpp	2017-01-02 01:03:56.000000000 +0100
+++ DevIL/src-IL/src/il_nvidia.cpp	2026-09-27 18:32:53.896075430 +0200
@@ -18,7 +18,6 @@
 
 #ifdef IL_USE_DXTC_NVIDIA
 #include <nvtt/nvtt.h>
-#include <nvcore/Memory.h>
 
 using namespace nvtt;
 
@@ -67,6 +66,9 @@ struct ilOutputHandlerMem : public nvtt:
 	{
 		// ignore.
 	}
+	virtual void endImage()
+	{
+	}
 	virtual bool writeData(const void *data, int size)
 	{
 		memcpy(Temp, data, size);
@@ -156,6 +158,9 @@ struct ilOutputHandlerFile : public nvtt
 	{
 		// ignore.
 	}
+	virtual void endImage()
+	{
+	}
 	virtual bool writeData(const void *data, int size)
 	{
 		if (iwrite(data, 1, size) == size)
